### ISSUE TITLE
Allow to generate SSL links and not try run LetsEncrypt on TrackDomains when work behind Proxy

### ISSUE THREAD

**Issue by dragoangel** [postalhq/postal#1129]
When Postal is setup behind reverse proxy people already can have correctly setup for SSL.
At same time Postal will fail to get LetsEncrypt HTTP validation as it will be go to RevProxy and end up there.
I think additional option is to add `fast_server.skip_letsencrypt=true` will be cool.
This will allow skip request correct cert and still issue correct https:// links inside email.

There is already have option to use fast_server.cert and fast_server.key so it not an issue to even without letsencrypt behind proxy has correct end-to-end ssl. As:
`https:proxy -> http-request set host header to correct track domain and validate cert which issued for it and stored at /opt/postal/config/fast_server.cert-> https:fastserver`
As option Postal UI with enabled `fast_server.skip_letsencrypt=true` can check that his link correctly work over SSL and only after this allow to use this track domain.

Also as idea at `/opt/postal/app/lib/postal/fast_server/client.rb` : `ssl_context.ssl_version` and `ssl_context.ciphers` put to config as they already done for smtp_server.

**Comment by dragoangel** [comment]
Also want to say that `fast_server.proxy_protocol=true` missing this part https://github.com/lucaspiller/postal/commit/928a2b8f034f92f9bad1a44e4079ac0ca8759c77 which will fix display of IPs behind proxy

**Comment by willpower232** [comment]
When you say reverse proxy, have you put the fast server behind nginx or do you have a separate proxy like a load balancer?

Unfortunately, the fast server is designed to respond to web requests directly so is a bit more challenging to proxy.

**Comment by dragoangel** [comment]
I have HAproxy, and it works fine with FastServer. I provide correct X-Forwarded-For and applied manually patch I mentioned before. All works except LetsEncrypt, as my HAproxy also use LetsEncrypt for all certs. Of course I doesn't need this certs to be requested, as my HAproxy are client facing, so I disable SSL, but all links now inserted with http. My HAproxy of course will forward all to https, but this no so cool =)

**Comment by willpower232** [comment]
I see. I don't know much about HAproxy but unless you were to send all the traffic for the click tracking domains to Postal so you didn't handle those SSL certificates in HAproxy, I don't think there is a quick solution here unless someone makes the code changes.

**Comment by dragoangel** [comment]
Yes, this is the case. If I can forward all traffic from one IP I will simply NAT it.

**Comment by dragoangel** [comment]
This issue not resolved and generally needed, why you closed it?

**Comment by willpower232** [comment]
this problem is very infrastructure specific and I don't believe has been reported before so unless you or someone with this setup makes the desired changes in a PR, I don't think this will happen any time soon

**Comment by dragoangel** [comment]
I understand your point, but still I think that it better to have this issue open if possible.

**Comment by masihfathi** [comment]
@dragoangel 
Can you help me about HAProxy config for the postal? I use a simple config to just forward http/s and smtp port to separate postal server with same db but the problem here it doesn't forward requests properly and just show me a blank page in the haproxy server which point to my domain or sth needed In caddy config of postal servers but I can't figure it out
